src/main/java/game/Game.java

```diff
@@ -181,10 +234,32 @@ public final void update(final List<String> input) {
 		createList.clear();
 		destroyList.clear();
 		createList.clear();
+		if (!player.isAlive()) {
+			if (score <= highscore) {
+				startGame();
+			} else {
+				highscore = score;
+				writeHighscore();
+				gamemode = GAMEMODE_HIGHSCORESCREEN;
+			}
+		}
```

Comment:
you should move this to the "over" method since it is called when the player dies.